processor/eventfilter/eventfilter.go

```diff
@@ -69,3 +77,325 @@ func GetSupportedMessageEvents(destination *backendconfig.DestinationT) ([]strin
 	}
 	return nil, false
 }
+
+type AllowTransformerEventParams struct {
+	TransformerEvent      *transformer.TransformerEventT
+	SupportedMessageTypes []string
+}
+
+type EventParams struct {
+	MessageType string
+}
+
+type ConnectionModeFilterParams struct {
+	Destination      *backendconfig.DestinationT
+	SrcType          string
+	Event            *EventParams
+	DefaultBehaviour bool
+}
+
+func getMessageType(event *types.SingularEventT) string {
+	eventMessageTypeI := misc.MapLookup(*event, "type")
+	if eventMessageTypeI == nil {
+		pkgLogger.Error("Message type is not being sent for the event")
+		return ""
+	}
+	eventMessageType, isEventTypeString := eventMessageTypeI.(string)
+	if !isEventTypeString {
+		pkgLogger.Errorf("Invalid message type :%v, Type assertion failed", eventMessageTypeI)
```

Comment:
[minor] Should the severity be an Error here, since it is not under our control, should we make this into Warn? 

Plus more go idiomatic way of logging [nipik] 

```suggestion
		pkgLogger.Error("Message type is not being sent for the event")
		return ""
	}
	eventMessageType, isEventTypeString := eventMessageTypeI.(string)
	if !isEventTypeString {
		pkgLogger.Errorf("Invalid message type: type assertion failed: %v", eventMessageTypeI)
```

Reply:
Done